contracts/test/TestTransfer.sol

```diff
@@ -6,26 +6,29 @@ import { Types } from "../libs/Types.sol";
 import { Tx } from "../libs/Tx.sol";
 
 contract TestTransfer is Transfer {
+    event Return1(uint256);
+    event Return2(Types.ErrorCode);
+
     function checkSignature(
         uint256[2] memory signature,
         InvalidSignatureProof memory proof,
         bytes32 stateRoot,
         bytes32 accountRoot,
         bytes32 appID,
         bytes memory txs
-    ) public returns (Types.ErrorCode, uint256 operationCost) {
-        operationCost = gasleft();
-        return (
-            _checkSignature(
-                signature,
-                proof,
-                stateRoot,
-                accountRoot,
-                appID,
-                txs
-            ),
-            operationCost - gasleft()
+    ) public {
+        uint256 operationCost = gasleft();
+        Types.ErrorCode err = _checkSignature(
+            signature,
+            proof,
+            stateRoot,
+            accountRoot,
+            appID,
+            txs
         );
+
+        emit Return1(operationCost - gasleft());
+        emit Return2(err);
```

Comment:
Here's how we get gas costs and result together. The `Types.ErrorCode err` assignment might bias the result a little bit but should be bearable. cc @kilic